apps/meteor/server/models/raw/Rooms.ts

```diff
@@ -0,0 +1,1978 @@
+		return this.find(
+			{
+				usersCount: { $gt: 2 },
+				uids: { $in: uids },
+			},
+			options,
+		);
+	}
+
+	find1On1ByUserId(userId: string, options: FindOptions<IRoom> = {}): FindCursor<IRoom> {
+		return this.find(
+			{
+				uids: userId,
+				usersCount: 2,
+			},
+			options,
+		);
+	}
+
+	findByCreatedOTR(): FindCursor<IRoom> {
+		return this.find({ createdOTR: true });
+	}
+
+	// UPDATE
+	addImportIds(_id: IRoom['_id'], importIds: string[]): Promise<UpdateResult> {
+		const query: Filter<IRoom> = { _id };
+
+		const update: UpdateFilter<IRoom> = {
+			$addToSet: {
+				importIds: {
+					$each: importIds,
+				},
+			},
+		};
+
+		return this.updateOne(query, update);
+	}
+
+	archiveById(_id: IRoom['_id']): Promise<UpdateResult> {
+		const query: Filter<IRoom> = { _id };
+
+		const update: UpdateFilter<IRoom> = {
+			$set: {
+				archived: true,
+			},
+		};
+
+		return this.updateOne(query, update);
+	}
+
+	unarchiveById(_id: IRoom['_id']): Promise<UpdateResult> {
+		const query: Filter<IRoom> = { _id };
+
+		const update: UpdateFilter<IRoom> = {
+			$set: {
+				archived: false,
+			},
+		};
+
+		return this.updateOne(query, update);
+	}
+
+	setNameById(_id: IRoom['_id'], name: IRoom['name'], fname: IRoom['fname']): Promise<UpdateResult> {
+		const query: Filter<IRoom> = { _id };
+
+		const update: UpdateFilter<IRoom> = {
+			$set: {
+				name,
+				fname,
+			},
+		};
+
+		return this.updateOne(query, update);
+	}
+
+	incMsgCountAndSetLastMessageById(
+		_id: IRoom['_id'],
+		inc = 1,
+		lastMessageTimestamp: IRoom['lm'],
+		lastMessage: IRoom['lastMessage'],
+	): Promise<UpdateResult> {
+		const query: Filter<IRoom> = { _id };
+
+		const update: UpdateFilter<IRoom> = {
+			$set: {
+				lm: lastMessageTimestamp,
+				...(lastMessage ? { lastMessage } : {}),
+			},
+			$inc: {
+				msgs: inc,
+			},
+		};
+
+		return this.updateOne(query, update);
+	}
+
+	incUsersCountById(_id: IRoom['_id'], inc = 1): Promise<UpdateResult> {
+		const query: Filter<IRoom> = { _id };
+
+		const update: UpdateFilter<IRoom> = {
+			$inc: {
+				usersCount: inc,
+			},
+		};
+
+		return this.updateOne(query, update);
+	}
+
+	// 4
+	incUsersCountNotDMsByIds(ids: Array<IRoom['_id']>, inc = 1): Promise<Document | UpdateResult> {
+		const query: Filter<IRoom> = {
+			_id: {
+				$in: ids,
+			},
+			t: { $ne: 'd' },
+		};
+
+		const update: UpdateFilter<IRoom> = {
+			$inc: {
+				usersCount: inc,
+			},
+		};
+
+		return this.updateMany(query, update);
+	}
+
+	setLastMessageById(_id: IRoom['_id'], lastMessage: IRoom['lastMessage']): Promise<UpdateResult> {
+		const query: Filter<IRoom> = { _id };
+
+		const update: UpdateFilter<IRoom> = {
+			$set: {
+				lastMessage,
+			},
+		};
+
+		return this.updateOne(query, update);
+	}
+
+	async resetLastMessageById(_id: IRoom['_id'], lastMessage: IRoom['lastMessage'] | undefined): Promise<UpdateResult> {
```

Comment:
```suggestion
	async resetLastMessageById(_id: IRoom['_id'], lastMessage: IRoom['lastMessage']): Promise<UpdateResult> {
```
`lastMessage` can already be undefined